app/src/main/python/gogdl/dl/workers/task_executor.py

```
@@ -0,0 +1,366 @@
+    """Download worker function that runs in a thread"""
+    session = requests.session()
+    
+    while True:
+        # Check for cancellation signal before processing next task
+        try:
+            import builtins
+            flag_name = f'GOGDL_CANCEL_{game_id}'
+            if hasattr(builtins, flag_name) and getattr(builtins, flag_name, False):
+                session.close()
+                return  # Exit worker thread if cancelled
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)